### PR TITLE
En_Tab

### PR DESCRIPTION
**Pull request by Kelebek1**
Before opening this PR, ensure the following:
- `./format.sh` was run to apply standard formatting.
- `make` successfully builds a matching ROM.
- No new compiler warnings were introduced during the build process.
    - Can be verified locally by running `tools/warnings_count/check_new_warnings.sh`
- New variables & functions should follow standard naming conventions.
- Comments and variables have correct spelling.
---

### CODE REVIEW

File: src/overlays/actors/ovl_En_Tab/z_en_tab.c [hunk 1]
```diff
@@ -31,75 +65,552 @@ const ActorInit En_Tab_InitVars = {
...
+            sp30 = arg2;
+        } else {
+            sp30 = arg2 * 0.5f;
+        }
+        if ((Math_Vec3f_DistXZ(&sp44, &sp38) < sp30) && (ABS_ALT(BINANG_SUB(sp2E, this->actor.shape.rot.y)) < 0x38E0)) {
+            ret = true;
+        }
+
+        if (DECR(this->unk_322) == 0) {
+            this->unk_338 ^= 1;
```
Comment (isghj5) [inline src/overlays/actors/ovl_En_Tab/z_en_tab.c:293]:
```
        if (this->unk_338 != 0) {
...
            this->unk_338 ^= 1;
```
Hmm, if we're XORing against a static 1 this is a toggle? Should we treat it as boolean and use 
```
        if (this->unk_338) {
```

File: src/overlays/actors/ovl_En_Tab/z_en_tab.c [hunk 2]
```diff
@@ -31,75 +65,552 @@ const ActorInit En_Tab_InitVars = {
...
+void EnTab_Init(Actor* thisx, GlobalContext* globalCtx) {
+    EnTab* this = THIS;
+
+    ActorShape_Init(&this->actor.shape, 0.0f, ActorShadow_DrawCircle, 14.0f);
+    SkelAnime_InitFlex(globalCtx, &this->skelAnime, &object_tab_Skel_007F78, NULL, this->jointTable, this->morphTable,
+                       20);
+    this->unk_32C = -1;
+    func_80BE05BC(this, 0);
+    Collider_InitAndSetCylinder(globalCtx, &this->collider, &this->actor, &sCylinderInit);
+    CollisionCheck_SetInfo2(&this->actor.colChkInfo, DamageTable_Get(0x16), &sColChkInfoInit);
```
Comment (isghj5) [inline src/overlays/actors/ovl_En_Tab/z_en_tab.c:504]: Musing: This is confusing but is also present in multiple other decompiled dialogue heavy actors:

```
CollisionCheck_SetInfo2(&this->actor.colChkInfo, DamageTable_Get(0x16), &sColChkInfoInit);
```
Why would a dialogue NPC like this have a DamageTable?

**Review by isghj5 — APPROVED** [review]
Nothing more than musings really

File: src/overlays/actors/ovl_En_Tab/z_en_tab.c [hunk 3]
```diff
@@ -31,75 +65,552 @@ const ActorInit En_Tab_InitVars = {
...
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Tab/func_80BE04E0.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Tab/func_80BE0590.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Tab/func_80BE05BC.s")
+static CollisionCheckInfoInit2 sColChkInfoInit = { 0, 0, 0, 0, MASS_IMMOVABLE };
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Tab/func_80BE0620.s")
+ActorAnimationEntryS D_80BE1AD0[] = { { 0x06000758, 1.0f, 0, -1, 0, 0 }, { 0x060086AC, 1.0f, 0, -1, 0, 0 } };
```
Comment (AngheloAlf) [inline src/overlays/actors/ovl_En_Tab/z_en_tab.c:90]: Don't forget to use the generated symbols from the object
```suggestion
ActorAnimationEntryS D_80BE1AD0[] = { { &object_tab_Anim_000758, 1.0f, 0, -1, 0, 0 }, { &object_tab_Anim_0086AC, 1.0f, 0, -1, 0, 0 } };
```

File: src/overlays/actors/ovl_En_Tab/z_en_tab.c (hunk 1 above)
Comment (AngheloAlf) [reply src/overlays/actors/ovl_En_Tab/z_en_tab.c:293]: Looks like this value is only used as a boolean, so it makes sense.

**Review by Kenix3 — APPROVED** [review]

**Pull request merged**